fuels-contract/src/contract.rs

```diff
@@ -136,8 +141,32 @@ impl Contract {
         );
         inputs.push(self_contract_input);
 
+        let spendables = wallet
+            .get_spendable_coins(&AssetId::default(), 10)
+            .await
+            .unwrap();
+        for coin in spendables {
+            let input_coin = Input::coin(
+                UtxoId::from(coin.utxo_id),
+                coin.owner.into(),
+                coin.amount.0,
+                AssetId::default(),
+                0,
+                0,
+                vec![],
+                vec![],
+            );
+
+            inputs.push(input_coin);
+        }
+
+        // let self_contract_output = Output::contract(0, Bytes32::zeroed(), Bytes32::zeroed());
+        // outputs.push(self_contract_output);
+
         let self_contract_output = Output::contract(0, Bytes32::zeroed(), Bytes32::zeroed());
         outputs.push(self_contract_output);
+        let change_output = Output::change(wallet.address(), 0, AssetId::default());
+        outputs.push(change_output);
```

Comment:
Specifically, `idx as u8 + 1` when creating the output is incorrect: it's assuming that the inputs being pushed are starting from 0, but they're not since you first push some coin input(s) to pay fees.

Reply:
Yup, since @vnepveu is now pushing these coin inputs we'll have to keep track of the number of inputs added at this stage, and then update this `idx as u8 + 1` to start from that previous number.

Reply:
I think I got it, will update soon!